### ISSUE TITLE
editor modelines should be a tidy error

### ISSUE THREAD

**Issue by edunham** [servo/servo#10719]
Somewhere in https://github.com/servo/servo/blob/master/python/tidy/servo_tidy/tidy.py we should check for editor detritus like 
```
-*- mode: modename; var: value; … -*-
# vim: set expandtab:
```

see https://www.gnu.org/software/emacs/manual/html_node/emacs/Specifying-File-Variables.html#Specifying-File-Variables, http://vim.wikia.com/wiki/Modeline_magic , and take a look through servo for any other editors' leftovers to catch if they exist.

As of https://github.com/servo/servo/pull/10723, we actually want to keep the modelines in `/tests/wpt`. Naive modeline checking is present but disabled in https://github.com/servo/servo/pull/10721/commits/93c3a6ee71a27c431e1292b5127d8d96aad1fb44; it needs to be extended to ignore checks within a specific directory just like other functions in the script. 

Note that the PR to fix this will need to include or follow removal of modelines from the rest of the project.

**Comment by edunham** [comment]
```
13:23:30 < aneeshusa> right, I forgot about those, I've never been a fan of editor modelines in code
13:23:36 <@edunham> larsberg: speaking of which, should editor residue lint as an error?
13:23:50 < aneeshusa> +1 yes please
13:24:11 <@larsberg> yeah, I think that makes sense, too
```

**Comment by Wafflespeanut** [comment]
fixed by #10723, #10786
